### ISSUE TITLE
Bug: Plays only the first file

### ISSUE THREAD

**Issue by ExXxtr3me** [Vito0912/abs_flutter#66]
The latest version is playing only the first file of an audiobook for me. Not sure if worked before because the last audiobook was only one file. Select the next files manually also doesn't work.

**Comment by Vito0912** [comment]
@ExXxtr3me Is the item openable at all?
Currently I cannot open a newly added test item with multiple tracks at all. So it would help if it's because of something special to this book

**Comment by ExXxtr3me** [comment]
It seems to be working for some files but not all.

**Comment by Vito0912** [comment]
Yes, I think I’ll need to rewrite the entire API section. My initial plan was to use OpenAPI, which I am currently implementing. However, Dart doesn’t support the `oneOf` operation, which allows a route to return multiple different classes. This lack of support makes debugging very challenging, especially with the current deserialization approach.

This is particularly problematic given the complexity of the ABS data, which has many variations. For example, sometimes `series` is an array, other times it’s a single object, and occasionally it’s divided into separate fields like `seriesName`.

I expect this will be an ongoing process, and until I find a solution, I may not be able to fully implement this. I’ll try setting up the first API route, but I'm not sure what kind of complications will follow.

**Comment by ExXxtr3me** [comment]
I do not really understand what that means but it seems you do and that's more important. 😁

**Comment by Vito0912** [comment]
With the latest release, I was able to stabilize the API, ensuring that all items are displayed. If an item isn’t displayed, there should be a clear log message explaining why, so I can fix it.

I also added support for multiple files. However, the playback duration doesn't update across files, so when moving to the second file, the time resets to 0, and progress restarts. Additionally, seeking is currently limited to the active file only.

I’ll definitely look into this issue further.
